### DISCUSSION TITLE
Need help with workspace numbering

### DISCUSSION THREAD

**Discussion by TheChosenOne1984** [polybar/polybar#2998] [Q&A]
If I open workspaces 1 to 9, and do a `xprop -root _NET_DESKTOP_NAMES` workspace 2 is missing and 9 appears twice.
`_NET_DESKTOP_NAMES(UTF8_STRING) = "1", "9", "3", "4", "5", "6", "7", "8", "9", ""`

Also, closing for example workspace 5 leads to the situation, that _NET_DESKTOP_NAMES returns this
`_NET_DESKTOP_NAMES(UTF8_STRING) = "1", "9", "3", "4", "5", "6", "7", "8", ""`

Polybar reflects exactly the status of _NET_DESKTOP_NAMES. But I want Polybar to Label the Workspace with the Number I use to access it. I'm using i3wm and if I press super+2, I want workspace 2 to appear, not 9. Or if I have workspaces from 1 to 9 open and close workspace 5, I want Polybar to display Workspaces 1 to 4 and 6 to 9.

Can someone give me a hint how to archive this?
Thank you!

**Comment by patrick96** [comment]
In your workspace labels, you can use `%name%` to display the workspace name (from `_NET_DESKTOP_NAMES`) and you can use `%index%` to display the workspace number. If you are using the i3 module, it will take the number reported by i3, for the xworkspaces module it will simply give you the position in the workspace list.

Whether this corresponds to your keyboard shortcuts, no idea.
